src/axios-har-tracker.ts

```diff
@@ -1,112 +1,126 @@
-          bodySize: JSON.stringify(resp.data).length,
-          redirectURL: '',
+        this.newEntry.request = {
+          method: config.method,
+          url: config.url,
+          httpVersion: version,
+          cookies: this.getCookies(fullCookie),
+          headers: [],
+          queryString: this.getParams(config.params),
           headersSize: -1,
-          content: {
-            size: JSON.stringify(resp.data).length,
-            mimeType: resp.headers['content-type'] ? resp.headers['content-type'] : 'text/plain',
-            text: JSON.stringify(resp.data)
-          },
-          cache: {},
-          timings: {
-            blocked: -1,
-            dns: -1,
-            ssl: -1,
-            connect: -1,
-            send: 10,
-            wait: 10,
-            receive: 10,
-            _blocked_queueing: -1
-          }
+          bodySize: -1
         };
-        const enteriesContent = Object.assign({}, newEntry);
-        generatedHar.log.entries.push(enteriesContent);
-        return resp;
+        return config;
+      },
+      error => {
+        return Promise.reject(error);
+      }
+    );
+
+    this.axios.interceptors.response.use(
+      async resp => {
+        if (resp) {
+          this.newEntry.response = {
+            status: resp.status,
+            statusText: resp.statusText,
+            headers: this.getHeaders(resp.headers),
+            startedDateTime: new Date(resp.headers.date),
+            time: resp.headers['request-duration'] = Math.round(
+              process.hrtime(resp.headers['request-startTime'])[0] * 1000 +
+                process.hrtime(resp.headers['request-startTime'])[1] / 1000000
+            ),
+            httpVersion: `HTTP/${resp.request.res.httpVersion}`,
+            cookies: this.getCookies(JSON.stringify(resp.config.headers['Cookie'])),
+            bodySize: JSON.stringify(resp.data).length,
+            redirectURL: '',
+            headersSize: -1,
+            content: {
+              size: JSON.stringify(resp.data).length,
+              mimeType: resp.headers['content-type'] ? resp.headers['content-type'] : 'text/plain',
+              text: JSON.stringify(resp.data)
+            },
+            cache: {},
+            timings: {
+              blocked: -1,
+              dns: -1,
+              ssl: -1,
+              connect: -1,
+              send: 10,
+              wait: 10,
+              receive: 10,
+              _blocked_queueing: -1
+            }
+          };
+          return resp;
+        }
+      },
+      error => {
+        return Promise.reject(error);
       }
-    },
-    error => {
-      return Promise.reject(error);
-    }
-  );
+    );
+  }
 
-  function transformObjectToArray(obj) {
+  resetNewEntry () {
+    this.newEntry = {}
+  }
+
+  public getGeneratedHar() {
+    const enteriesContent = Object.assign({}, this.newEntry);
+    this.generatedHar.log.entries.push(enteriesContent);
```

Comment:
Pushing new entry to entries array should happen after every request. Right now it's pushed only once when getGenerateHar is invoked and it will contain only last newEntry object